src/Microsoft.IdentityModel.Tokens/Json/JsonSerializerPrimitives.cs

```diff
@@ -1129,6 +1129,10 @@ public static void WriteObject(ref Utf8JsonWriter writer, string key, object obj
 
                 writer.WriteEndObject();
             }
+            else if ((typeof(byte[])).IsAssignableFrom(objType))
```

Comment:
Shouldn't this be `is byte[] bytes`? Is it even possible to inherit from `byte[]`?

Reply:
> Shouldn't this be `is byte[] bytes`? Is it even possible to inherit from `byte[]`?

What's there makes sense to me because `obj` is being cast and `IsAssignableFrom` checks if that can be done, whereas `is` will check if the type inherits.

Reply:
The main thing is `IsAssignableFrom` involves reflection whereas a simple `is` check doesn't. The reflection makes sense I think for other types but not for primitives.

Reply:
+` to use `is`
`is` seems to be more versatile as you can use it with expressions. It does seem to be equivalent to `IsAssignableFrom`. 

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast#is-operator
https://learn.microsoft.com/en-us/dotnet/api/system.type.isassignablefrom?view=net-8.0#returns
https://stackoverflow.com/a/15853224

Reply:
Either way worked, but yes, is is preferable here.